test/parallel/xyk-pallet.API.sellAsset.test.ts

```diff
@@ -110,36 +141,39 @@ describe("xyk-pallet - Sell assets tests: SellAsset Errors:", () => {
         expect(eventResponse.data).toEqual(xykErrors.NoSuchPool);
       });
 
-    await validateUnmodified(
-      firstCurrency,
-      secondCurrency,
-      testUser1,
-      pool_balance_before
-    );
-  });
+    await testUser1.refreshAmounts(AssetWallet.AFTER);
 
-  test("Try sell more assets than owned", async () => {
-    [firstCurrency, secondCurrency] = await Assets.setupUserWithCurrencies(
-      testUser1,
-      [defaultCurrecyValue, defaultCurrecyValue.add(new BN(1))],
-      sudo
+    const secondCurrencyValueAfterSelling =
+      testUser1.getAsset(secondCurrency)?.amountAfter.free!;
+    const thirdCurrencyValueAfterSelling =
+      testUser1.getAsset(thirdCurrency)?.amountAfter.free!;
+
+    expect(secondCurrencyValueBeforeSelling).bnEqual(
+      secondCurrencyValueAfterSelling
     );
-    await testUser1.addMGATokens(sudo);
-    await testUser1.refreshAmounts(AssetWallet.BEFORE);
-    await createPool(
-      testUser1.keyRingPair,
-      firstCurrency,
-      first_asset_amount,
-      secondCurrency,
-      second_asset_amount.div(new BN(2))
+    expect(thirdCurrencyValueBeforeSelling).bnEqual(
+      thirdCurrencyValueAfterSelling
     );
 
+    const pool_balance = await getBalanceOfPool(secondCurrency, thirdCurrency);
+    expect([pool_balance_before[0], pool_balance_before[1]]).collectionBnEqual(
+      pool_balance
+    );
+
+    const balance = await getBalanceOfPool(thirdCurrency, secondCurrency);
+    expect([pool_balance_before[0], pool_balance_before[1]]).collectionBnEqual([
+      balance[1],
+      balance[0],
+    ]);
+  });
+
+  test("Try sell more assets than owned", async () => {
```

Comment:
same again, if you remove the tokens setup, you need to update the user assets on every test.

Reply:
Fixed